### PR TITLE
Fix 2967

### PR DESCRIPTION
**Pull request by antgonza**
GUI interaction:

How a non-owner/admin will see the deprecation message:
deprecated

Note that this will not prevent adding the artifacts to an analyses but now the analyses will have an auto-added new column: `qiita_prep_deprecated`.

@justinshaffer, could you take a look too on the functionality?

### CODE REVIEW

**Comment by antgonza** [comment]
Well, to have the study list not show artifacts that were deprecated via the prep information was much harder than expected. Mainly because there was no artifact/preparation table. 

Note that all performance testing was done in qiita-rc. Anyway, I initially tried to avoid having to create a new table by using a pgsql function but the retrieval time incremented by ~5 times (*):
```python
In [13]: %timeit x = generate_study_list_new(User('antoniog@ucsd.edu'), 'public')                                                                                                                                                                                                                 
20.1 s ± 39.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

However, adding the new table actually slightly improves retrieval time:
```python
In [7]: %timeit x = generate_study_list(User('antoniog@ucsd.edu'), 'public')                                                                                                                                                                                                                      
4.03 s ± 11.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [26]: %timeit x = generate_study_list_new(User('antoniog@ucsd.edu'), 'public')                                                                                                                                                                                                                 
3.86 s ± 14.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

File: qiita_pet/templates/study_ajax/prep_summary.html [hunk 1]
```diff
@@ -133,6 +133,41 @@
     }
   }
 
+
+  /*
+   *
+   * Updates deprecated flag in the prep information <fieldset>
+   *
+   * @param prep_id the prep template id
+   *
```
Comment (ElDeveloper) [inline qiita_pet/templates/study_ajax/prep_summary.html]: looks like `new_value` is missing here.

File: qiita_pet/templates/study_ajax/prep_summary.html [hunk 2]
```diff
@@ -133,6 +133,41 @@
...
+   */
+  function deprecate_preparation(prep_id, new_value) {
+    var status = 'deprecate';
+    console.log(new_value)
+    if (!new_value){
+      status = 'DE-deprecate'
+    }
+    if(confirm("Are you sure you want to " + status + " this preparation?")) {
+      if(confirm("Are you _really_ sure you want to " + status + " this preparation?")) {
```
Comment (ElDeveloper) [inline qiita_pet/templates/study_ajax/prep_summary.html]: Two "alerts" seems like a little too much in my opinion. Is there a strong motivation for this?

File: qiita_pet/templates/study_ajax/prep_summary.html [hunk 3]
```diff
@@ -396,6 +436,12 @@ <h4>
         <a class="btn btn-default" href="{% raw qiita_config.portal_dir %}/download_ebi_accessions/experiments/{{prep_id}}"><span class="glyphicon glyphicon-download-alt"></span> EBI experiment accessions</a>
       {% end %}
       {% if editable %}
+        <br/>
+        {% if deprecated %}
+          <a class="btn btn-warning" onclick="deprecate_preparation({{prep_id}}, false);"><span class="glyphicon glyphicon-pushpin"></span> Remove Deprecate</a>
```
Comment (ElDeveloper) [inline qiita_pet/templates/study_ajax/prep_summary.html]:
```suggestion
          <a class="btn btn-warning" onclick="deprecate_preparation({{prep_id}}, false);"><span class="glyphicon glyphicon-pushpin"></span> Remove Deprecation</a>
```

File: qiita_db/util.py [hunk 4]
```diff
@@ -1520,17 +1499,25 @@ def generate_study_list(user, visibility):
...
-                info['artifact_biom_ids'] = []
-                if info['aids_with_deprecation'] is not None:
-                    for x in info['aids_with_deprecation']:
-                        # f1-2 are the default names given by pgsql
-                        if not x['f2']:
-                            info['artifact_biom_ids'].append(x['f1'])
-                del info['aids_with_deprecation']
-
-                if info['preparation_data_types'] is None:
-                    info['preparation_data_types'] = []
```
Comment (ElDeveloper) [inline qiita_db/util.py]: Is this no longer needed?

File: qiita_db/metadata_template/prep_template.py [hunk 5]
```diff
@@ -480,6 +480,29 @@ def study_id(self):
             qdb.sql_connection.TRN.add(sql, [self.id])
             return qdb.sql_connection.TRN.execute_fetchlast()
 
+    @property
+    def deprecated(self):
+        with qdb.sql_connection.TRN:
+            sql = """SELECT deprecated FROM qiita.prep_template
+                     WHERE {0} = %s""".format(self._id_column)
+            qdb.sql_connection.TRN.add(sql, [self._id])
+            return qdb.sql_connection.TRN.execute_fetchlast()
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/metadata_template/prep_template.py]: bool:
```python
In [1]: from qiita_db.metadata_template.prep_template import PrepTemplate                                                                                                                                                                                                                         

In [2]: PrepTemplate(1).deprecated                                                                                                                                                                                                                                                                
Out[2]: False

In [3]: type(PrepTemplate(1).deprecated)                                                                                                                                                                                                                                                          
Out[3]: bool
```

File: qiita_db/util.py (hunk 4 above)
Comment (antgonza) [reply qiita_db/util.py]: This is correct, now the preparation data types is retrieved together with the artifact info, lines 1502 and 1511 below.

File: qiita_pet/templates/study_ajax/prep_summary.html (hunk 2 above)
Comment (antgonza) [reply qiita_pet/templates/study_ajax/prep_summary.html]: Just that in sensitive things we always add two but in this case it's pretty easy to correct so I will leave only one.

File: qiita_db/metadata_template/prep_template.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ElDeveloper) [reply qiita_db/metadata_template/prep_template.py]: Thanks!

File: qiita_pet/templates/study_ajax/prep_summary.html (hunk 2 above)
Comment (ElDeveloper) [reply qiita_pet/templates/study_ajax/prep_summary.html]: My bad, if two confirmations are widespread in Qiita then I rather be consistent than go for a personal preference. Thanks eitherway! 👍

**Comment by antgonza** [comment]
Nah, I agree that 2 confirmations were too much and normally the double confirmation is for those things that can't be undone (like delete a full study) ... and here a user will just need to click again, so 2 confirmations were a lot ... thank you for checking.

**Comment by ElDeveloper** [comment]
Thanks @antgonza!

**Pull request merged**